lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -158,6 +102,56 @@ sub run {
         Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, ( $params->{ipv6} ) ? ( 1 ) : ( 0 ) );
     }
 
+    my %methods = Zonemaster::Engine->all_methods;
+
+    # BASIC methods are always run
+    $counter_for_progress_indicator{planned} = {
+        'Basic::basic00' => 1,
+        'Basic::basic01' => 1,
+        'Basic::basic02' => 1,
+        'Basic::basic03' => 1,
+        'Basic::basic04' => 1,
```

Comment:
Is it good and necessary to refer to specific test cases here? Which test cases that are run even not included in profile can change.

Basic03 is normally not run. There is no "TEST_CASE_END basic03" message and HAS_NAMESERVER_NO_WWW_A_TEST does not come from test case basic03.

Reply:
> Is it good and necessary to refer to specific test cases here?

Since progress is defined as `progress = (nbr finished test cases ) / (nbr planned test cases)`, I need to compute the number of planned test. Getting them from the profile is not enough as you noted.

> Which test cases that are run even not included in profile can change.

True, this is something to keep in mind. Maybe Engine should be updated in a way so that it makes it easier to get the list of planned test cases. But this would be part of a future work.

> Basic03 is normally not run.

True that Basic03 is normally not run. I thought it won't alter the progress computation too much to keep it anyways in the number of planned test. I think it's better that the following inequality always holds: `(nbr planned test cases) >= (nbr finished test cases)`.

> There is no "TEST_CASE_END basic03" message ...

This is not what I found while working on this. "TEST_CASE_END" is properly defined ([master](https://github.com/zonemaster/zonemaster-engine/blob/master/lib/Zonemaster/Engine/Test/Basic.pm#L126) and [develop](https://github.com/zonemaster/zonemaster-engine/blob/develop/lib/Zonemaster/Engine/Test/Basic.pm#L125)) and used ([master](https://github.com/zonemaster/zonemaster-engine/blob/master/lib/Zonemaster/Engine/Test/Basic.pm#L523) and [develop](https://github.com/zonemaster/zonemaster-engine/blob/develop/lib/Zonemaster/Engine/Test/Basic.pm#L522)).

> ...  and HAS_NAMESERVER_NO_WWW_A_TEST does not come from test case basic03.

Sorry I don't understand what you mean here. Could you elaborate please?

Reply:
> > There is no "TEST_CASE_END basic03" message ...
> 
> This is not what I found while working on this. "TEST_CASE_END" is properly defined ([master](https://github.com/zonemaster/zonemaster-engine/blob/master/lib/Zonemaster/Engine/Test/Basic.pm#L126) and [develop](https://github.com/zonemaster/zonemaster-engine/blob/develop/lib/Zonemaster/Engine/Test/Basic.pm#L125)) and used ([master](https://github.com/zonemaster/zonemaster-engine/blob/master/lib/Zonemaster/Engine/Test/Basic.pm#L523) and [develop](https://github.com/zonemaster/zonemaster-engine/blob/develop/lib/Zonemaster/Engine/Test/Basic.pm#L522)).

What I meant was that since basic03 is not run in most cases, the "TEST_CASE_END basic03" message is not outputted in those cases. Of course it is outputted if it is run.

> > ...  and HAS_NAMESERVER_NO_WWW_A_TEST does not come from test case basic03.
> 
> Sorry I don't understand what you mean here. Could you elaborate please?

The HAS_NAMESERVER_NO_WWW_A_TEST tag is outputted outside the basic03 code.

Reply:
> The HAS_NAMESERVER_NO_WWW_A_TEST tag is outputted outside the basic03 code.

The progress computation does not rely on this message. Only on "TEST_CASE_END".

> What I meant was that since basic03 is not run in most cases, the "TEST_CASE_END basic03" message is not outputted in those cases. Of course it is outputted if it is run.

Indeed, having a proper way to list all planned test would be great but outside the scope of this PR. TBH even though BASIC03 is not always run I think this is fine to keep it listed as part of always planned test cases to avoid cases where `(nbr of finished tests) > (nbr of planned tests)` (and a progress > 100).